studio/components/LanguageSelector.tsx

```diff
@@ -0,0 +1,151 @@
+import { Box, Button, Card, Checkbox, Flex, Text, useTheme } from "@sanity/ui";
+import React from "react";
+import { PatchEvent, set } from "sanity";
+
+import {
+  Language,
+  supportedLanguages,
+} from "internationalization/supportedLanguages";
+
+interface LanguageSelectorProps {
+  value?: Language[];
+  onChange: (event: PatchEvent) => void;
+}
+
+const colorMap = {
+  dark: {
+    selected: "#29356a",
+    default: "rgba(19, 20, 27, 0.5)",
+  },
+  light: {
+    selected: "#E3F2FD",
+    default: "white",
+  },
+};
+
+const LanguageSelector = ({ value = [], onChange }: LanguageSelectorProps) => {
+  const theme = useTheme();
+  const prefersDark = theme.sanity.v2?.color._dark ?? false;
+
+  // Get the currently set default language
+  const currentDefaultLanguage = value.find((lang) => lang.default)?.id || null;
+
+  const handleLanguageSelection = (lang: Language) => {
+    const isSelected = value.some((item) => item.id === lang.id);
+    // Prevent deselecting the last remaining language
+    if (isSelected && value.length === 1) {
+      console.log("Cannot deselect the last remaining language.");
+      return; // Exit early if there's only one language selected
+    }
+
+    const updatedValue = isSelected
+      ? value.filter((item) => item.id !== lang.id) // Deselect language
+      : [...value, { ...lang, default: false }]; // Select language
+
+    const newDefaultLanguage = getNewDefaultLanguage(
+      updatedValue,
+      currentDefaultLanguage,
+      lang,
+    );
+
+    const finalValue = updatedValue.map((item) => ({
+      ...item,
+      default: item.id === newDefaultLanguage,
+    }));
+
+    onChange(PatchEvent.from(set(finalValue)));
+  };
+
+  const handleDefaultSetting = (lang: Language) => {
+    const updatedValue = value.map((item) => ({
+      ...item,
+      default: item.id === lang.id, // Set selected language as default
+    }));
+    onChange(PatchEvent.from(set(updatedValue)));
+  };
+
+  const getBackgroundColor = (lang: Language) => {
+    const isSelected = value.some((item) => item.id === lang.id);
+    const theme = prefersDark ? "dark" : "light";
+
+    return isSelected ? colorMap[theme].selected : colorMap[theme].default;
+  };
+
+  return (
+    <Flex direction="column" gap={3}>
+      {supportedLanguages.map((lang) => (
+        <Card
+          key={lang._key}
+          padding={4}
+          radius={2}
+          shadow={1}
+          style={{
```

Comment:
Should we not avoid inline styling?

Reply:
I've been strict for plain html elements, but not for sanity ui. 🤔  we could do a cleanup when we look at the css variables for colors in studio as well